### ISSUE TITLE
Custom transform plugin is not running on mac

### ISSUE THREAD

**Issue by bilbeyt** [pylint-dev/pylint#8825]
### Bug description

I cant run transform plugin properly and you can see the below codes.

```plugins/pylint.py
from typing import TYPE_CHECKING

from astroid import MANAGER, Call

if TYPE_CHECKING:
    from pylint.lint import PyLinter

print("plugin loaded")

def register(linter: "PyLinter") -> None:
    """This required method auto registers the checker during initialization.

    :param linter: The linter to register the checker to.
    """
    print('register')

def transform(node):
    print('transform ran')

MANAGER.register_transform(Call, transform)
```

```test.py
def func():
    pass

func()
```

I am using `pylint --load-plugins plugins.pylint test.py` to test and i can only see register log. My friend with linux machine can see also transform log.

### Command used

```shell
pylint --load-plugins plugins.pylint test.py
```

### Pylint output

```shell
register
```

### Expected behavior

register
transform ran

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.10.9
```

### OS / Environment

MacOS 12.6.7

**Comment by jacobtylerwalls** [comment]
> and i can only see register log

Ah, sorry, this should indicate that the plugin was found. Still can't reproduce on Mac though.

**Comment by bilbeyt** [comment]
hello @jacobtylerwalls thanks for response. i can use a screen recording if it would help or if you have some questions, i can answer you.

**Comment by jacobtylerwalls** [comment]
Yeah, we will need more instructions to reproduce. Can you share the layout of your package and the contents of sys.path?

A video could be a good last resort also.

**Comment by bilbeyt** [comment]
hey, you can check this branch https://github.com/beamer-bridge/beamer/tree/fix/tx_typings for content and package layout, and for sys.path:
```
['/Users/tolga/.virtualenvs/beamer/bin',
 '/Users/tolga/.virtualenvs/beamer/lib/python3.10/site-packages/_pdbpp_path_hack',
 '/Users/tolga/.pyenv/versions/3.10.9/lib/python310.zip',
 '/Users/tolga/.pyenv/versions/3.10.9/lib/python3.10',
 '/Users/tolga/.pyenv/versions/3.10.9/lib/python3.10/lib-dynload',
 '',
 '/Users/tolga/.virtualenvs/beamer/lib/python3.10/site-packages',
 '/Users/tolga/projects/beamer']
```

**Comment by DanielNoord** [comment]
You are not installing the plugin in https://github.com/beamer-bridge/beamer/pull/1991/files which means it is unavailable and therefore won't run. You'll need to make sure the plugin is loaded in your `pylintrc` file.

**Comment by bilbeyt** [comment]
I am using command line option to load it, but i also tried having it in pylintrc. nothing changes really.

**Comment by DanielNoord** [comment]
> I am using command line option to load it, but i also tried having it in pylintrc. nothing changes really.

I'm probably overlooking something, but where are you using the CLI option to load it in that PR?

**Comment by jacobtylerwalls** [comment]
I think the PR was only for showing the package layout.

**Comment by bilbeyt** [comment]
Yes, i just wanted. to show layout with the PR/branch, but locally i tried --load-plugins or used pyrintrc

**Comment by DanielNoord** [comment]
@bilbeyt It would be good to show how you intend to use this in the PR because then we can help debug you. It's quite hard to debug an issue on somebody else's machine while it works on your own :sweat_smile:

**Comment by bilbeyt** [comment]
@DanielNoord i have added plugins.pylint to my pylintrc and this will be used like this normally.

**Comment by DanielNoord** [comment]
@bilbeyt Could you remove the breakpoint so I can see what the output of the run is?

**Comment by bilbeyt** [comment]
> @bilbeyt Could you remove the breakpoint so I can see what the output of the run is?

done

**Comment by bilbeyt** [comment]
@DanielNoord temporarily solved mypy issues by ignoring them so we can continue with pylint.

**Comment by DanielNoord** [comment]
Wait, the CI is running fine right? I thought we were debugging the CI, sorry! For your `Mac` issue I really don't know where to start. This works on my local Mac..

**Comment by jacobtylerwalls** [comment]
Feel free to reopen if you can provide enough instructions to reproduce the issue.
